crates/accelerate/src/convert_2q_block_matrix.rs

```diff
@@ -52,22 +52,36 @@ pub fn blocks_to_matrix(
         let op_matrix = op_matrix.as_array().into_faer_complex();
 
         let result = match q_list.as_slice() {
-            [0] => identity.kron(op_matrix.as_ref()),
-            [1] => op_matrix.kron(identity.as_ref()),
-            [1, 0] => change_basis_faer(op_matrix),
-            [] => Mat::<c64>::identity(4, 4),
-            _ => op_matrix.to_owned(),
+            [0] => Some(identity.kron(op_matrix.as_ref())),
+            [1] => Some(op_matrix.kron(identity.as_ref())),
+            [1, 0] => Some(change_basis_faer(op_matrix)),
+            [] => Some(Mat::<c64>::identity(4, 4)),
+            _ => None,
         };
 
         let aux = matrix.clone();
```

Comment:
Since `matrix` gets overwritten anyways to a new value, would it make more sense to create a zeros-matrix for the first arg and avoid this clone? Then you can directly pass `matrix` as the third arg. According to the docs, because the fourth arg is `None`, the operation is:

> acc = beta * lhs * rhs if alpha is None (in this case, the preexisting values in acc are not read, so it is allowed to be a view over uninitialized values if E: Copy),

Reply:
Good observation! However, I think we can't avoid the `clone()`, unfortunately. The matrix multiplication is inside a for loop, and we use the result of the previous iteration to calculate the current one. Moreover, `matrix` is the final result of the function and gets modified outside the for loop too